src/main/kotlin/io/kuzzle/sdk/coreClasses/http/Route.kt

```diff
@@ -0,0 +1,178 @@
+    "controller",
+    "action",
+    "meta",
+    "volatile",
+)
+
+class Route {
+    private val partType: PartType
+    private val routeParts: ArrayList<RoutePart>
+    private var staticURL: String = ""
+    private var baseURL: String
+    private var verb: String
+
+    private constructor(verb: String, baseUrl: String, routeParts: ArrayList<RoutePart>) {
+        this.verb = verb.uppercase()
+        this.baseURL = baseUrl
+        this.routeParts = routeParts
+        val hasOnePartTemplated = routeParts.indexOfFirst { it.type == PartType.TEMPLATE } > -1
+        this.partType = if (hasOnePartTemplated) PartType.TEMPLATE else PartType.STATIC
+        if (! hasOnePartTemplated) {
+            staticURL = routeParts.joinToString("") {
+                it.value
+            }
+        }
+    }
+
+    /**
+     * Given map containing the properties of the request, the buildRequest method
+     * will create an instance of HttpRequest which contains the information needed to make
+     * the request later on.
+     *
+     * The method has to:
+     * - Generate the URL using the properties of the request if there are part of the URL that are templated
+     * - Generate the query string
+     * - Generate the list of headers
+     */
+    fun buildRequest(request: KuzzleMap): HttpRequest {
+        val headers = request.optMap("headers", KuzzleMap())
+
+        val queryArgs = KuzzleMap()
+        for (key: String? in request.keys) {
+            // Skip if a key or value is null
+            if (key == null || request[key] == null) {
+                continue
+            }
+
+            when (key) {
+                "jwt" -> headers["authorization"] = "Bearer ${request["jwt"]}"
+                "volatile" -> {
+                    headers["x-kuzzle-volatile"] = request["volatile"]
+                }
+                "headers" -> headers.putAll(request.optMap("headers", KuzzleMap()))
+                "body" -> {
+                    if (verb == "GET") {
+                        queryArgs.putAll(request.optMap("body", KuzzleMap()))
+                    }
+                }
+                else -> {
+                    if (! PayloadProperties.contains(key)) {
+                        queryArgs[key] = request[key]
+                    }
+                }
+            }
+        }
+
+        /**
+         * Build the query string
+         */
+        val queryString: String = queryArgs.keys.filterNotNull().joinToString("&") {
+            val encodedKey = URLEncoder.encode(it, "utf-8")
+
+            if (queryArgs.isArrayList(it)) {
+                val value = queryArgs.getArrayList(it)!!.joinToString(",")
+                "$encodedKey=${URLEncoder.encode(value, "utf-8")}"
+            } else if (queryArgs.optBoolean(it, false) == true) {
+                encodedKey
+            } else {
+                val value = if (queryArgs.isMap(it)) JsonSerializer.serialize(queryArgs.getMap(it)!!) else queryArgs[it].toString()

```

Comment:
Same comment for objects

Reply:
Same has before, I just copied how the SDK-Javascript behaves

You can see this here: https://github.dev/kuzzleio/sdk-javascript/blob/a8e94a1f57a8b794c805865f7e879b1bec0401e0/src/protocols/Http.ts#L289-L309

Reply:
I changed this to skip over maps and arrays and only encode boolean, integer, and strings in the url

Reply:
I tried to remove this behaviour and suddenly document mGet and other stopped working, it seems that the SDK-JS is doing this on purpose otherwise something does not work, so I will continue to copie how the SDK-JS behaves even though it's not ideal.